### PR TITLE
Port 2.16.0 to Main

### PR DESCRIPTION
**Pull request by JillieBeanSim**
## Proposed changes

port changes from 2.16.0 to main branch

## Release Notes

Milestone: n/a

Changelog: n/a

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: packages/zowe-explorer/src/uss/USSTree.ts [hunk 1]
```diff
@@ -23,7 +23,16 @@
...
+import {
+    Gui,
+    IZoweTree,
+    IZoweTreeNode,
+    IZoweUSSTreeNode,
+    NodeInteraction,
+    ValidProfileEnum,
+    PersistenceSchemaEnum,
+    ZosEncoding,
+} from "@zowe/zowe-explorer-api";
```
[the remark this answers is not in the document]
Comment (t1m0thyj) [reply packages/zowe-explorer/src/uss/USSTree.ts:35]: This import shouldn't be unused - I think there are some changes missing from the `openWithEncoding` function on maintenance branch:
https://github.com/zowe/zowe-explorer-vscode/blob/c68a110eef0d90bf8c83a3352a228f2a6eb9b545/packages/zowe-explorer/src/uss/USSTree.ts#L968-L981

File: packages/zowe-explorer/src/uss/init.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@
 import * as vscode from "vscode";
 import * as ussActions from "./actions";
 import * as refreshActions from "../shared/refresh";
-import { Gui, IZoweUSSTreeNode, IZoweTreeNode } from "@zowe/zowe-explorer-api";
+import { Gui, IZoweUSSTreeNode, IZoweTreeNode, ZosEncoding } from "@zowe/zowe-explorer-api";
```
[the remark this answers is not in the document]
Comment (t1m0thyj) [reply packages/zowe-explorer/src/uss/init.ts:16]: This import shouldn't be unused (same as previous comment) - I think there are some changes missing from the `maintenance` branch:
https://github.com/zowe/zowe-explorer-vscode/blob/c68a110eef0d90bf8c83a3352a228f2a6eb9b545/packages/zowe-explorer/src/uss/init.ts#L209-L214

**Review by t1m0thyj — CHANGES_REQUESTED** [review]

**Comment by JillieBeanSim** [comment]
sorry got confused with date changed comparison, hopefully this is correct now @t1m0thyj

**Review by zFernand0 — APPROVED** [review]
LGTM, thanks @JillieBeanSim 😋

File: packages/zowe-explorer/__tests__/__unit__/shared/utils.unit.test.ts [hunk 3]
```diff
@@ -1085,41 +1101,200 @@ describe("Shared utils unit tests - function promptForEncoding", () => {
     });
 });
 
-describe("Shared utils unit tests - function confirmForUnsavedDoc", () => {
```
Comment (t1m0thyj) [inline packages/zowe-explorer/__tests__/__unit__/shared/utils.unit.test.ts:1088]: I believe this `describe` block should be kept - we want to keep both sets of tests for `confirmForUnsavedDoc` and `initializeFileOpening`.

File: packages/zowe-explorer/src/dataset/DatasetTree.ts [hunk 4]
```diff
@@ -1533,21 +1525,11 @@ export class DatasetTree extends ZoweTreeProvider implements IZoweTree<IZoweData
         }
     }
 
-    public async openWithEncoding(node: IZoweDatasetTreeNode): Promise<void> {
-        const encoding = await promptForEncoding(node);
-
-        const docResult = await confirmForUnsavedDoc(node);
```
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/DatasetTree.ts:1539]: The logic on this line and below related to `confirmForUnsavedDoc` should be kept.

File: packages/zowe-explorer/src/uss/USSTree.ts [hunk 5]
```diff
@@ -964,25 +965,18 @@ export class USSTree extends ZoweTreeProvider implements IZoweTree<IZoweUSSTreeN
...
 
-    public async openWithEncoding(node: IZoweUSSTreeNode): Promise<void> {
-        const ussApi = ZoweExplorerApiRegister.getUssApi(node.getProfile());
-        let taggedEncoding: string;
-        if (ussApi.getTag != null) {
-            taggedEncoding = await ussApi.getTag(node.fullPath);
-        }
-        const encoding = await promptForEncoding(node, taggedEncoding !== "untagged" ? taggedEncoding : undefined);
-
-        const docResult = await confirmForUnsavedDoc(node);
```
Comment (t1m0thyj) [inline packages/zowe-explorer/src/uss/USSTree.ts:975]: The logic on this line and below related to `confirmForUnsavedDoc` should be kept.

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
There seem to be a few encoding changes related to https://github.com/zowe/zowe-explorer-vscode/pull/2911 that were lost. Otherwise LGTM.

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @JillieBeanSim!

**Review by traeok — APPROVED** [review]
LGTM, thanks @JillieBeanSim!

**Pull request merged**